client/pcr.go

```diff
@@ -99,64 +98,22 @@ func ReadAllPCRs(rw io.ReadWriter) ([]*pb.PCRs, error) {
 	return allPcrs, nil
 }
 
-// SealCurrent seals data to the current specified PCR selection.
-type SealCurrent struct{ tpm2.PCRSelection }
-
-// SealTarget predicatively seals data to the given specified PCR values.
-type SealTarget struct{ Pcrs *pb.PCRs }
-
 // SealOpts specifies the PCR values that should be used for Seal().
-type SealOpts interface {
-	PCRsForSealing(rw io.ReadWriter) (*pb.PCRs, error)
-}
-
-// PCRsForSealing read from TPM and return the selected PCRs.
-func (p SealCurrent) PCRsForSealing(rw io.ReadWriter) (*pb.PCRs, error) {
-	if len(p.PCRSelection.PCRs) == 0 {
-		panic("SealCurrent contains 0 PCRs")
-	}
-	return ReadPCRs(rw, p.PCRSelection)
-}
-
-// PCRsForSealing return the target PCRs.
-func (p SealTarget) PCRsForSealing(_ io.ReadWriter) (*pb.PCRs, error) {
-	if len(p.Pcrs.GetPcrs()) == 0 {
-		panic("SealTarget contains 0 PCRs")
-	}
-	return p.Pcrs, nil
-}
-
-// CertifyCurrent certifies that a selection of current PCRs have the same value when sealing.
-// Hash Algorithm in the selection should be CertifyHashAlgTpm.
-type CertifyCurrent struct{ tpm2.PCRSelection }
-
-// CertifyExpected certifies that the TPM had a specific set of PCR values when sealing.
-// Hash Algorithm in the PCR proto should be CertifyHashAlgTpm.
-type CertifyExpected struct{ Pcrs *pb.PCRs }
-
-// CertifyOpts determines if the given PCR value can pass certification in Unseal().
-type CertifyOpts interface {
-	CertifyPCRs(rw io.ReadWriter, certified *pb.PCRs) error
+type SealOpts struct {
+	// Current seals data to the current specified PCR selection.
+	Current tpm2.PCRSelection
+	// Target predictively seals data to the given specified PCR values.
+	Target *pb.PCRs
 }
 
-// CertifyPCRs from CurrentPCRs will read PCR values from TPM and compare the digest.
-func (p CertifyCurrent) CertifyPCRs(rw io.ReadWriter, pcrs *pb.PCRs) error {
-	if len(p.PCRSelection.PCRs) == 0 {
-		panic("CertifyCurrent contains 0 PCRs")
-	}
-	current, err := ReadPCRs(rw, p.PCRSelection)
-	if err != nil {
-		return err
-	}
-	return internal.CheckSubset(current, pcrs)
-}
-
-// CertifyPCRs will compare the digest with given expected PCRs values.
-func (p CertifyExpected) CertifyPCRs(_ io.ReadWriter, pcrs *pb.PCRs) error {
-	if len(p.Pcrs.GetPcrs()) == 0 {
-		panic("CertifyExpected contains 0 PCRs")
-	}
-	return internal.CheckSubset(p.Pcrs, pcrs)
+type UnsealOpts struct {
+	// CertifyCurrent certifies that a selection of current PCRs have the same
+	// value when sealing.
+	// Hash Algorithm in the selection should be CertifyHashAlgTpm.
```

Comment:
Can we also update the docs on `CertifyHashAlgTpm` to be:
```go
/// CertifyHashAlgTpm is the hard-coded algorithm that must be used with UnsealOpts.
```

Reply:
done.